### PR TITLE
Change MockResponse body to Source.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
Sigh. More hanging tests.

File: mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockResponse.java [hunk 1]
```diff
@@ -130,15 +128,20 @@ public MockResponse removeHeader(String name) {
...
+    return body;
   }
 
-  Source getBodySource() {
-    return bodyStream != null ? Okio.source(bodyStream) : getBody();
+  /**
+   * The number of bytes in {@linkplain #getBody() the body}. Note: this is not synonymous with
+   * {@code Content-Length} (e.g., in the case of chunked body).
+   */
+  public long getBodySize() {
```
Comment (swankjesse) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockResponse.java]: Looks like we only ever compare this to 0. Perhaps instead we could just have a null response from body source? The fact that this doesn't line up with `Content-Length` is ugly.

File: mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java [hunk 2]
```diff
@@ -855,15 +855,16 @@ private void writeResponse(SpdyStream stream, MockResponse response) throws IOEx
...
-      if (body == null) body = new Buffer();
-      boolean closeStreamAfterHeaders = body.size() > 0 || !response.getPushPromises().isEmpty();
+
+      Source body = response.getBody();
+      long bodySize = response.getBodySize();
+      boolean closeStreamAfterHeaders = bodySize > 0 || !response.getPushPromises().isEmpty();
       stream.reply(spdyHeaders, closeStreamAfterHeaders);
       pushPromises(stream, response.getPushPromises());
-      if (body.size() > 0) {
+      if (bodySize > 0) {
```
Comment (swankjesse) [inline mockwebserver/src/main/java/com/squareup/okhttp/mockwebserver/MockWebServer.java]: `if (body != null) ` ?

**Comment by swankjesse** [comment]
LGTM

**Comment by JakeWharton** [comment]
I went full-buffer which means you can stream anymore. Do we care?

The problem is that some tests re-enqueue the same `MockResponse` instance for multiple requests. This worked if and only if you were using a `Buffer`/`byte[]` but not if you were specifying an `InputStream` (which nobody was internally). Do we need to support one-shot streaming responses?

**Comment by swankjesse** [comment]
Let's kill one-shot streaming responses. They were there for testing large payloads.

**Pull request merged**